root/utility/formatUserDate.js

```diff
@@ -6,35 +6,152 @@
  * http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
-import moment from 'moment';
-import trim from 'lodash/trim';
+import parseIsoDate from './parseIsoDate';
 
-import 'moment-strftime';
-import 'moment-timezone';
+const formatterCache = new Map();
 
-function formatUserDate(
-  user: ?(CatalystUserT | SanitizedEditorT),
-  dateString: string,
-  options?: {dateOnly?: boolean},
+/*
+ * This maps the `strftime` patterns we use to `Intl.DateTimeFormat` options.
+ *
+ * The object values below are in the form [typeName, options]. The typeName
+ * identifies the exact subset of information we need to satisfy the pattern.
+ * This is necessary because `DateTimeFormat` implementations are only
+ * required to support a limited number of format combinations. [1] To work
+ * around that, we use `formatToParts` and extract the specific part we want
+ * using typeName as a key.
+ *
+ * Where typeName is null, we use `format` instead of `formatToParts`, since
+ * we want the entire string. (In those cases, the specified options should
+ * comprise a format combination that's always supported.)
+ *
+ * [1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/
+ *     Global_Objects/DateTimeFormat
+ */
+
+const patterns = {
+  '%A': ['weekday', {weekday: 'long'}],
+  '%B': ['month', {month: 'long'}],
+  '%H': ['hour', {hour: '2-digit', hour12: false}],
+  '%M': ['minute', {minute: '2-digit'}],
+  '%S': ['second', {second: '2-digit'}],
+  '%X': [null, {
+    hour: 'numeric',
+    minute: 'numeric',
+    second: 'numeric',
+  }],
+  '%Y': ['year', {year: 'numeric'}],
+  '%Z': ['timeZoneName', {timeZoneName: 'short'}],
+  '%a': ['weekday', {weekday: 'short'}],
+  '%b': ['month', {month: 'short'}],
+  '%c': [null, {
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+    month: 'numeric',
+    second: 'numeric',
+    year: 'numeric',
+  }],
+  '%d': ['day', {day: '2-digit'}],
+  '%e': ['day', {day: 'numeric'}],
+  '%m': ['month', {month: '2-digit'}],
+  '%x': [null, {
+    day: 'numeric',
+    month: 'numeric',
+    year: 'numeric',
+  }],
+};
+
+let HAS_INTL_DATETIMEFORMAT = true;
+
+try {
+  const now = new Date();
+  formatDateUsingPattern(now, '%a', 'America/Chicago', 'en-US');
+  formatDateUsingPattern(now, '%c', 'America/Chicago', 'en-US');
+} catch (e) {
+  HAS_INTL_DATETIMEFORMAT = false;
+}
+
+function formatDateUsingPattern(
+  date: Date,
```

Comment:
Minor af, but shouldn't this have a space before the paren? Or is it different if all arguments are on the next lines?

Reply:
There's a space before the opening parenthese for anonymous functions (`function ()`), but only because there's always a space after `function`, so it's consistent in both cases. There's never a space after the function name, though (compare `git grep 'function [0-9A-z] \+('` with `git grep 'function [0-9A-z]\+('`).